website/docs/guides/upgrade.html.markdown

```diff
@@ -0,0 +1,158 @@
+---
+layout: "nsxt"
+page_title: "VMware NSX Terraform Provider NSX upgrade support"
+description: |-
+  The VMware NSX Terraform Provider NSX upgrade support
+---
+
+# NSX Provider for NSX Upgrade
+
+NSX Terraform Provider offers support for [upgrading NSX components](https://docs.vmware.com/en/VMware-NSX/4.1/upgrade/GUID-E04242D7-EF09-4601-8906-3FA77FBB06BD.html).
+
+The provider support consists of several resources and data sources which can perform a complete upgrade cycle when applied correctly.
+The NSX upgrade process upgrades the manager components of NSX, Edge appliances, and the NSX bits on the compute hosts.
+
+
+## Upgrade process steps
+
+### Preparation for upgrade
+
+Upgrade preparation consists of several actions, and is performed using [nsxt_upgrade_prepare](../resources/upgrade_prepare.html.markdown):
+* Uploading the upgrade and precheck bundles.
+* Acceptance of the user agreement.
+* Reporting of failed pre-checks.
+
+```hcl
+resource "nsxt_upgrade_prepare" "prepare_res" {
+  upgrade_bundle_url    = "http://url-to-upgrade-bundle.mub"
+  precheck_bundle_url   = "http://url-to-precheck-bundle.pub"
+  accept_user_agreement = true
+}
+```
+
+**NOTE:** Acceptance of the license agreement is required to initiate the upgrade process.
+**NOTE:** Precheck bundle is usable only while upgrading a NSX manager of version v4.1.1 and above.
+
+### Acknowledgement of upgrade issues and warnings
+
+The preparation stage may result with various warnings which would halt the upgrade process. These warnings can be 
+suppressed with the `nsxt_upgrade_precheck_acknowledge` resource.
+
+For example:
+```
+There are unacknowledged warnings in precheck: [backupOperationCheck], please address these errors from NSX or using nsxt_upgrade_precheck_acknowledge resource
+```
+
+The resource uses the `precheck_ids` to conclude which prechecks are safe to ignore for this upgrade cycle.
+The `nsxt_upgrade_precheck_acknowledge` resource requires the `target_version` from the `nsxt_upgrade_prepare` resource 
+to make sure that the acknowledged warnings are from the correct execution, and to enforce execution of the 
+`nsxt_upgrade_precheck_acknowledge` resource after preparation is complete.
+
+The `nsxt_upgrade_prepare_ready` data source verifies that the preparation phase of the upgrade was concluded 
+successfully and enable the execution of the upgrade. 
+
+```hcl
+resource "nsxt_upgrade_precheck_acknowledge" "test" {
+  precheck_ids   = ["backupOperationCheck", "pUBCheck"]
+  target_version = nsxt_upgrade_prepare.test.target_version
```

Comment:
Fore completeness, you might want to show also the nsxt_upgrade_prepare resource.
And if you're actually referring to the previous one, it might worth replacing `nsxt_upgrade_prepare.test` with `nsxt_upgrade_prepare.prepare_res`

Reply:
Wouldn't it become cumbersome if each HCL example will include all the proceeding DS/resources?

Reply:
Agreed, let's not make things too difficult for us